tests/unit/Rules/EqualsTest.php

```diff
@@ -13,61 +13,53 @@
 
 namespace Respect\Validation\Rules;
 
-use PHPUnit\Framework\TestCase;
+use Respect\Validation\Test\RuleTestCase;
 use stdClass;
 
 /**
- * @group  rule
+ * @group rule
+ *
  * @covers \Respect\Validation\Rules\Equals
- * @covers \Respect\Validation\Exceptions\EqualsException
+ *
+ * @author Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ * @author Henrique Moody <henriquemoody@gmail.com>
  */
-class EqualsTest extends TestCase
+final class EqualsTest extends RuleTestCase
 {
     /**
-     * @dataProvider providerForEquals
+     * {@inheritdoc}
      */
-    public function testInputEqualsToExpectedValueShouldPass($compareTo, $input): void
+    public function providerForValidInput(): array
     {
-        $rule = new Equals($compareTo);
-
-        self::assertTrue($rule->validate($input));
+        return [
+            [new Equals('foo'), 'foo'],
+            [new Equals([]), []],
+            [new Equals(new stdClass()), new stdClass()],
+            [new Equals(10), '10'],
+            [new Equals(10), 10.0],
+        ];
     }
 
     /**
-     * @dataProvider providerForNotEquals
+     * {@inheritdoc}
      */
-    public function testInputNotEqualsToExpectedValueShouldPass($compareTo, $input): void
+    public function providerForInvalidInput(): array
     {
-        $rule = new Equals($compareTo);
-
-        self::assertFalse($rule->validate($input));
+        return [
+            [new Equals('foo'), ''],
+            [new Equals('foo'), 'bar'],
+        ];
     }
 
     /**
-     * @expectedException \Respect\Validation\Exceptions\EqualsException
-     * @expectedExceptionMessage "24" must equal 42
+     * @test
      */
-    public function testShouldThrowTheProperExceptionWhenFailure(): void
+    public function shouldPassCompareToParameterToException(): void
```

Comment:
When the rule passes parameters to exceptions a test like that should be created.